### PR TITLE
Updating initializer handling module

### PR DESCRIPTION
**Pull request by leogermani**
#### Changes proposed in this Pull Request:

This PR changes how the VideoPress package is initialized by the Jetpack plugin and starts initializing it from the VideoPress plugin.

It also handles the videopress module state depending on which plugin is installed.

* If Jetpack plugin is active: the module will be enabled or disabled depending on the choice the user makes on the Settings page or in the My Jetpack page to activate or not the module
* If the VideoPress standalone plugin is active: the module will always be considered active

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

NA

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Start with a site with Jetpack active and connected

We are going to run some tests in the UI and some tests on the WP Shell CLI (`wp shell`)

* Go to Jetpack > Settings, search for video, and make sure videopress is disabled
* On WP shell: `( new Automattic\Jetpack\Modules() )->is_active('videopress');` should return false
* `Automattic\Jetpack\VideoPress\Status::is_active();` should also return false.
* On the Settings page, activate videopress
* Run the same two commands above and they should return true this time
* On the Settings page, deactivate videopress
* Activate the VideoPress plugin
* Run the same two commands again and they should return true.
* Go to Jetpack Settgins and search for video. Verify that videopress is displayed as active and that the toggle to disable it is not active (you can not disable videopress)

Captura de tela de 2022-08-04 17-20-27

* Deactivate Jetpack plugin
* Go to WP CLI and run the two commands again. They should still return true while videopress plugin is active

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202708227705595

### CODE REVIEW

File: projects/packages/videopress/src/class-status.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	 * @return boolean
+	 */
+	public static function is_standalone_plugin_active() {
+		$possible_folders = array(
+			'videopress', // monorepo.
+			'jetpack-videopress', // production.
+			'jetpack-videopress-dev', // Jetpack Beta.
+		);
+		foreach ( $possible_folders as $folder ) {
+			if ( is_plugin_active( $folder . '/jetpack-videopress.php' ) ) {
```
Comment (retrofox) [inline projects/packages/videopress/src/class-status.php]: Getting a Fatal here:

```
[04-Aug-2022 21:36:26 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function Automattic\Jetpack\VideoPress\is_plugin_active() in /usr/local/src/jetpack-monorepo/projects/packages/videopress/src/class-status.php:50
```

```suggestion
			if ( Jetpack::is_plugin_active( $folder . '/jetpack-videopress.php' ) ) {
```

could ⬆️ work?

**Comment by leogermani** [comment]
This should mean that this code is being called too early. I got this at
some point but fixed it. I'll double check to see what I missed. Thanks for
testing

Em qui., 4 de ago. de 2022 18:39, Damián Suárez ***@***.***>
escreveu:

File: projects/packages/videopress/src/class-status.php (hunk 1 above)
Comment (leogermani) [reply projects/packages/videopress/src/class-status.php]: I ran all the tests again now and I dont see this error. In what situation exactly it happened to you?

Comment (leogermani) [reply projects/packages/videopress/src/class-status.php]: What we could do instead if the problem persists is `class_exists( 'Jetpack_VideoPress_Plugin' ); `

Comment (leogermani) [reply projects/packages/videopress/src/class-status.php]: well, I could not reproduce it for some reason, but I was seeing the tests failing, so I changed the way we do the check in https://github.com/Automattic/jetpack/pull/25385/commits/76f39e8f3748ece4ffade5cc357ac0665519357d

Comment (leogermani) [reply projects/packages/videopress/src/class-status.php]: it should work now

**Comment by retrofox** [comment]
When I deactivate VideoPress plugin, I get this Fatal error:

```
[05-Aug-2022 12:16:30 UTC] PHP Fatal error: 
Uncaught Error: Class "Automattic\Jetpack\VideoPress\Module_Control" not found
in /usr/local/src/jetpack-monorepo/projects/packages/videopress/src/class-initializer.php:44
```

**Comment by retrofox** [comment]
> If the VideoPress standalone plugin is active: the module will always be considered active

I'm not able to deactivate the VideoPress standalone plugin because of https://github.com/Automattic/jetpack/pull/25385#issuecomment-1206381139

**Comment by leogermani** [comment]
@retrofox can you double check your env? I dont see this error... in what situation exactly it happens?

**Comment by retrofox** [comment]
> @retrofox can you double check your env? I dont see this error... in what situation exactly it happens?

I've built all packages and not too much else. Could you provide basic instructions to set up the local dev with these changes? what should we do?

**Comment by leogermani** [comment]
Either test on a JN site or run `composer update` from inside the jetpack plugin and the videopress plugin folders. this should do the job

**Review by retrofox — APPROVED** [review]
- [x] Go to Jetpack > Settings, search for video, and make sure videopress is disabled

image

- [x] On WP shell: ( new Automattic\Jetpack\Modules() )->is_active('videopress'); should return false
- [x] Automattic\Jetpack\VideoPress\Status::is_active(); should also return false.

image

- [x] On the Settings page, activate videopress

image

- [x] Run the same two commands above and they should return true this time

image

- [x] On the Settings page, deactivate videopress
- [x] Activate the VideoPress plugin
- [x] Run the same two commands again and they should return true.
- [x] Go to Jetpack Settgins and search for video. Verify that videopress is displayed as active and that the toggle to disable it is not active (you can not disable videopress)

image

LGMT 🚀

**Comment by retrofox** [comment]
> Either test on a JN site or run `composer update` from inside the jetpack plugin and the videopress plugin folders. this should do the job

sorry. works fine now.

**Pull request merged**